airflow/cli/commands/remote_commands/dag_command.py

```diff
@@ -346,8 +353,14 @@ def get_dag_detail(dag: DAG) -> dict:
             dag_detail = _get_dagbag_dag_details(dag)
         return {col: dag_detail[col] for col in valid_cols}
 
+    def filter_dags_by_bundle():
+        """Filter DAGs based on the specified bundle name, if provided."""
+        if args.bundle_name:
+            return [dag for dag in dagbag.dags.values() if dag.get_bundle_name() == args.bundle_name]
+        return dagbag.dags.values()
```

Comment:
```suggestion
    def filter_dags_by_bundle(dags: list[DAG], bundle_name: str | None) -> list[DAG]:
        """Filter DAGs based on the specified bundle name, if provided."""
        if not bundle_name:
            return dags
        return [dag for dag in dags if dag.get_bundle_name() == bundle_name]
```

Let's pass in the dags and bundle name instead of relying on scope. Cleaner and easier to grok.

(note: this does result in a query per dag to go find the bundle name, but thats a short term problem until this is refactored to use the api anyways, so not going to worry about it)